internal/handlers/lokistack_create.go

```diff
@@ -66,9 +67,48 @@ func CreateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client) error
 		l.Info("Resource created", "resource", obj.GetName())
 	}
 
+	RegisterLokiSpecMetrics(stack.Spec)
+
 	return nil
 }
 
+// Registers metrics based on the LokiStack spec
+func RegisterLokiSpecMetrics(spec lokiv1beta1.LokiStackSpec) {
```

Comment:
FYI, `register` refers in prometheus metrics terminology to the one-time task to register the counters/gauges/histogram (e.g. as you do in `metrics.go`). I suggest to rename this function to `collectMetricsFor(spec lokiv1beta1.LokiStackSpec)`